manila/share/drivers/scality/driver.py

```diff
@@ -45,44 +72,157 @@
 class ScalityShareDriver(driver.ShareDriver):
     """Scality Ring driver for Manila.
 
-    Supports NFS through the sfused NFS connector.
+    Supports NFS through the Sfused NFS connector.
+    Supports SMB through the Sfused SMB connector.
     """
-    # Cli exit codes
-    EXPORT_NOT_FOUND = 10
-
-    ACCESS_EXISTS = 11
-
-    ACCESS_NOT_FOUND = 12
-
-    HAS_GRANTS = 13
 
-    # Driver version
     VERSION = '1.0'
 
-    NFS_PROTOCOL = 'NFS'
-
     def __init__(self, *args, **kwargs):
         super(ScalityShareDriver, self).__init__(False, *args, **kwargs)
         self.configuration.append_config_values(share_opts)
-        self._config_check()
-
-        self.export_ip = self.configuration.export_ip
-        self.ssh_pool = utils.SSHPool(
-            ip=self.configuration.export_management_host,
-            port=self.configuration.export_management_port,
-            conn_timeout=None,
-            login=self.configuration.management_user,
-            privatekey=self.configuration.ssh_key_path,
-        )
-
-    def _config_check(self):
-        """Validate configuration settings."""
-        log.debug('Validating configuration')
-        required = ('export_management_host', 'management_user', 'export_ip')
-        for config_item in required:
-            if getattr(self.configuration, config_item) is None:
-                msg = _("'%s' must be configured" % config_item)
-                raise exception.ManilaException(msg)
+
+        self._helpers = {}
+
+    def do_setup(self, context):
+        super(ScalityShareDriver, self).do_setup(context)
+
+        LOG.debug('Validating Scality Driver configuration')
+
+        msg = _("Configuration error: missing value for %(item)s. "
+                "Check the manila.conf file")
+
+        # Check generic configuration
+        for config_item in ('management_user', 'ssh_key_path'):
+            # Accept neither None nor ""
+            if not self.configuration.get(config_item):
```

Comment:
Agree with @sbjorn, rather use something like
```python
if self.configuration.get(config_item, '') == '':
```
Python's bool'ish behavior is odd once in a while.